### ISSUE TITLE
Flutter web do not set lang for screen readers

### ISSUE THREAD

**Issue by Leffe108** [flutter/flutter#98948]
## Steps to Reproduce
To reproduce as in the attached video you have to have Swedish as your locale in Windows 10. However, the test contains texts of 3 different locales and as web fails to communicate any of them, you should be able to reproduce it in some way regardless of the locale/language of your system. It very likely is possible to reproduce on Mac and Linux as well. I can reproduce it on an iPad with VoiceOver.

1. `flutter create bug_locale`
2. replace lib/main.dart with code sample
2. `flutter run -d chrome`
3. Turn on the screen reader. On windows 10 the shortcut is Win+Ctrl+Enter. If you use iPad/iOS you can ask Siri to turn on VoiceOver.
4. Press Narrator key + W to have it read out the contents of the window (Narrator key on my system is Caps Lock and I don't think I've changed the default). If you use an iPad/iOS swipe down with two fingers for it to read out the content.

**Expected results:**  
I expect that the screen reader is using the corresponding voice for the different texts. The texts them self try to convey this and in the sample code there are comments that state the expected language of each text. It is fine for the screen reader to use the system locale for annotations like window, button, link etc.

If I visit other websites like https://flutter.dev Windows Narrator is using an English voice to read out the content. (it uses the system default voice - Swedish - to announce annotations like "link", "image" etc., but content like "Flutter", "Multi-platform", "Development etc. is read out in English. )

On a technical side I expect the rendered dom tree to set `lang` attribute on the html or a high up node in body to the locale defined by MaterialApp and then leaf-content to have `lang` set based on the locale overrides I have made in the code. [BBC has a short but useful article that explains this](https://www.bbc.co.uk/accessibility/forproducts/guides/mobile/indicating-language/?fbclid=IwAR2XPXHnFKNm5jWRJH10NZjBwh0b_a5-Mu-0py2kaVNJuvzyGH4jj_V0-tY) - see the HTML sub-heading.

**Actual results:**
Windows Narrator uses Swedish voice all through the video. In the beginning it says "fönsterruta" which is swedish for window and that is expected and possible also using Swedish to announce the tab title "Flutter Demo". But the content starting with Hello World should be according to what is defined in dart.
https://user-images.githubusercontent.com/4630670/155220124-4fa13ead-caf8-4bea-bf7a-c2803babfa6a.mp4

For reference, here is Flutter.dev where it uses English for content:
https://user-images.githubusercontent.com/4630670/155220504-c67ce585-9ac1-4bf7-8910-ffa9221104ee.mp4

<details>
<summary>Code sample - lib/main.dart</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      supportedLocales: const [
        Locale('en', 'UK'),
      ],
      locale: const Locale('en', 'UK'),
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Hello World!'), // should be voiced with UK Engilsh voice (app default)
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text('Regular text without locale override. Should use UK voice.'), // should be vocied with UK English vocie (app default)
            Container(height: 20.0),
            const Text.rich(
              TextSpan(children: [
                TextSpan(text: 'UK text (no override). '), // should be vocied with UK English vocie (app default)
                TextSpan(text: 'UK text (using override). ', locale: Locale('en', 'UK')), // should be voiced with UK English vocie
                TextSpan(text: 'American text. ', locale: Locale('en', 'US')), // should be voiced with US English voice
                TextSpan(text: 'Dansk text. ', locale: Locale('da', 'DK')), // should be voiced with Danish vocie
              ]),
            ),
            Container(height: 20.0),
            Localizations.override(context: context,
              locale: const Locale('da', 'DK'),
              child: const Text('Localizations.override på dansk.'), // should be voiced with Danish voice
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

```
flutter analyze
Analyzing bug_locale...
No issues found! (ran in 4.6s)
```

```
 flutter doctor -v
[✓] Flutter (Channel stable, 2.10.2, on Microsoft Windows [Version 10.0.19044.1526], locale sv-SE)
    • Flutter version 2.10.2 at C:\Users\Leif\Programmering\Libraries\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 097d3313d8 (4 days ago), 2022-02-18 19:33:08 -0600
    • Engine revision a83ed0e5e3
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at K:\Android\android-sdks
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2022 17.0.6)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.0.32126.317
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 2021.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)

[✓] VS Code, 32-bit edition (version 1.64.2)
    • VS Code at C:\Program Files (x86)\Microsoft VS Code
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19044.1526]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 98.0.4758.82
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 98.0.1108.50

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

# Sidenote
I have on my todo to also report that the same sample app when built for iOS will also not set the locale for the app using setAccessibilityLanguage. So the system language is used for texts that have no override. Also the Localizations.override part doesn't work on iOS. The TextSpan override do work on iOS meaning that it is a Flutter Web bug that none of the locale is communicated to the browser. But there can be also framework wide bugs connected to some ways of communicating locale to screen readers.

**Comment by Leffe108** [comment]
I am able to reproduce the same results in Flutter Windows - no part is localized correctly by Windows Narrator.

https://user-images.githubusercontent.com/4630670/155224779-80b75b8a-d2e4-4418-b2a7-33c1b2ad8d03.mp4

**Comment by Leffe108** [comment]
Since I expect many readers to have en-US or en-UK as their system locale I provide also a demo code where app language is set to Swedish so you can hear that the screen reader uses the English system voice to read out the text and not the Swedish voice. If you know a bigger second language like French or so, maybe re-write to that to avoid the pitfall that Swedish voice is not installed on your system. Also it is easier for you to hear when it is right or wrong if you pick a language you know at least a little bit of that is different from your system language.

<details>
<summary>lib/main.dart - Swedish app language verison for developers with English system language</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      supportedLocales: const [
        Locale('sv', 'SE'),
      ],
      locale: const Locale('sv', 'SE'),
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Hej världen!'), // should be voiced with Swedish voice (app default)
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text('Normal text utan tvingande språk'), // should be vocied with Swedish vocie (app default)
            Container(height: 20.0),
            const Text.rich(
              TextSpan(children: [
                TextSpan(text: 'Svensk text (utan tvingande regel). '), // should be vocied with Swedish vocie (app default)
                TextSpan(text: 'Svensk text (med tvingande regel). ', locale: Locale('sv', 'SE')), // should be voiced with Swedish vocie (due to override)
                TextSpan(text: 'Dansk text. ', locale: Locale('da', 'DK')), // should be voiced with Danish vocie
              ]),
            ),
            Container(height: 20.0),
            Localizations.override(context: context,
              locale: const Locale('sv', 'SE'),
              child: const Text('Localizations.override på svenska.'), // should be voiced with Swedish voice
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

**Comment by maheshmnj** [comment]
Hi @Leffe108, thanks for filing the issue, I am able to reproduce the issue, The narrator uses the default voice to read texts with different locales. I did also try using flutter_localizations package but the output was same on both stable and the master channel.


https://user-images.githubusercontent.com/31410839/155293898-9c86e798-1422-48f5-b768-d7a11507a963.mp4

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter_localizations/flutter_localizations.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      localizationsDelegates: [GlobalMaterialLocalizations.delegate],
      restorationScopeId: 'app',
      title: 'Flutter Demo',
      supportedLocales: const [
        Locale('en', 'UK'),
      ],
      locale: const Locale('en', 'UK'),
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Hello World!'), // should be voiced with UK Engilsh voice (app default)
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text('Regular text without locale override. Should use UK voice.'), // should be vocied with UK English vocie (app default)
            Container(height: 20.0),
            const Text.rich(
              TextSpan(children: [
                TextSpan(text: 'UK text (no override). '), // should be vocied with UK English vocie (app default)
                TextSpan(text: 'UK text (using override). ', locale: Locale('en', 'UK')), // should be voiced with UK English vocie
                TextSpan(text: 'American text. ', locale: Locale('en', 'US')), // should be voiced with US English voice
                TextSpan(text: 'Dansk text. ', locale: Locale('da', 'DK')), // should be voiced with Danish vocie
              ]),
            ),
            Container(height: 20.0),
            Localizations.override(context: context,
              locale: const Locale('da', 'DK'),
              child: const Text('Localizations.override på dansk.'), // should be voiced with Danish voice
            ),
          ],
        ),
      ),
    );
  }
}
```
</details>


<details>
<summary>flutter doctor -v</summary>

```
[√] Flutter (Channel stable, 2.10.2, on Microsoft Windows [Version 10.0.19044.1526], locale en-US)    • Flutter version 2.10.2 at C:\flutter_sdk\stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 097d3313d8 (4 days ago), 2022-02-18 19:33:08 -0600
    • Engine revision a83ed0e5e3
    • Dart version 2.16.1
    • DevTools version 2.9.2

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.64.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.34.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19044.1526]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 98.0.4758.102
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 98.0.1108.56

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
```
[√] Flutter (Channel master, 2.11.0-0.0.pre.554, on Microsoft Windows [Version 10.0.19044.1526],
    locale en-US)
    • Flutter version 2.11.0-0.0.pre.554 at C:\flutter_sdk\master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 894ef49c7c (8 days ago), 2022-02-14 22:15:17 -0500
    • Engine revision 8714a2448f
    • Dart version 2.17.0 (build 2.17.0-105.0.dev)
    • DevTools version 2.10.0

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java      
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)    
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.64.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.34.0

[√] Connected device (4 available)
    • Windows (desktop)       • windows • windows-x64     • Microsoft Windows [Version
      10.0.19044.1526]
    • Windows (UWP) (desktop) • winuwp  • windows-uwp-x64 • 
    • Chrome (web)            • chrome  • web-javascript  • Google Chrome 98.0.4758.102
    • Edge (web)              • edge    • web-javascript  • Microsoft Edge 98.0.1108.56

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by Leffe108** [comment]
iOS video with Italian as system language (also posted in https://github.com/flutter/flutter/issues/99018 - both issues uses exact same code) 

iOS share the bug that MaterialApp locale and Localization.override text uses wrong voice. iOS does however correctly support TextSpan locale with screen readers.

https://user-images.githubusercontent.com/4630670/155397588-42c3f146-e605-4283-a118-d6e826e0d75c.mp4

**Comment by Leffe108** [comment]
@maheshmnj Thank you for triaging the issue. Perhaps also add the label [a: accessibility](https://github.com/flutter/flutter/labels/a%3A%20accessibility)?

**Comment by chunhtai** [comment]
It looks like this feature is not hooked up in web and desktop.

**Comment by TzviPM** [comment]
I've assigned this to myself and changed the priority to p3 to indicate that I'm actively working on it.

**Comment by jacobsimionato** [comment]
I believe Tzvi is not looking at the moment, so unassigning for now. This seems like a high priority issue - it would be great to ensure that screenreaders on all platforms use the correct language.

**Comment by MilosKarakas** [comment]
Hi @TzviPM , are there any updates on this issue? We have a working app for iOS/Android written in Flutter, with over 300k users in total, and this issue is constantly popping up in our feature requests. We have many visually impaired users which require screen readers in order to use our app. 

Thank you.

**Comment by Leffe108** [comment]
> Hi @TzviPM , are there any updates on this issue? We have a working app for iOS/Android written in Flutter, with over 300k users in total, and this issue is constantly popping up in our feature requests. We have many visually impaired users which require screen readers in order to use our app.
> 
> Thank you.

Thank you for bringing attention to screen reader issues in Flutter. This issue in particular however I have not been able to reproduce on iOS and Android. Only Web and Windows (I don't have access to Linux/Mac so those are untested for this issue).

This issue to recap is about `TextSpan(locale: )` not communicating locale override to screen readers. Something which in past research has been functional on iOS and Android. 

There is a nearby issue/feature request https://github.com/flutter/flutter/issues/99600 that affects iOS/Android.

Please clarify if you indeed have issues with `TextSpan(locale: )` not working on iOS/Android so this issue can be expanded, or if your issue is related to other APIs?

**Comment by Leffe108** [comment]
I can still reproduce this in flutter 3.7.11 as well as using flutter master (119744a63c08b82486d6202891843c8206aff5f0) using the code example of top post. I used iOS with Safari as test device.

**Comment by Piinks** [comment]
This is still an important issue, it looks like this affects multiple teams though. @cbracken or @loic-sharma maybe for the Windows portion, and maybe @yjbanov for web would know the best course to take for this

**Comment by gspencergoog** [comment]
Assigning one person to take a look at this for the platforms, as below:

  - @cbracken : macOS
  - @loic-sharma : Windows
  - @mdebbar : web
  - @jpnurmi : Linux

**Comment by Leffe108** [comment]
I have been informed in another issues that Localizations.override does by design not communicate locale to assistive tech. So that part I should probably remove from the OP post to make this issue more concise.

The issue here is that the TextSpan part that is documented to communicate locale to assistive tech (and does so on iOS) do not do it on desktop or web. Eg. there is likely missing support on the layer of communicating semantic locale to the assistive APIs of the platform.

For extending ability to declare assistive tech locale I have opened separate issues for that.

**Comment by loic-sharma** [comment]
@Leffe108 Do you have a sample app with steps to demonstrate Windows screen readers properly announcing text with overridden language? From my test using Chrome, it seems Windows does not have good support for overriding the language on text.

### My Chrome test

I created this test HTML and opened it in Chrome:

```html
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
  <title>Accessibility language demo</title>
</head>
<body>
  <p lang="fr">Bonjour narrateur comment vas-tu?</p>
  <ul>
    <li><a href="#a">Hello narrator how are you?</a></li>
    <li><a href="#b">Bonjour narrateur comment vas-tu ?</a></li>
    <li><a href="#c"><span lang="en">Hello narrator how are you?</span></a></li>
    <li><a href="#c"><span lang="fr">Bonjour narrateur comment vas-tu ?</span></a></li>
    <li><a href="#d"><span lang="en">Hello narrator </span><span lang="fr">comment vas-tu ?</span></a></li>
  </ul>
</body>
</html>
```

Using Accessibility Insights and Inspect.exe, I can see that Chrome sets the texts' `Culture` property to `0x9` for English and `0xC` for French.

Using NVDA, the text is announced in the desired language if I focus the text. However, NVDA announces the text using the incorrect language if I hover over it.

Using Windows Narrator, everything is announced in whatever language Narrator is configured to. If I use an English voice, everything is announced in English. If I use a French voice, everything is announced in French.

**Comment by Leffe108** [comment]
I just re-installed windows (upgraded to windows 11), and I found that I first could not reproduce the example I provided in OP to visit https://flutter.dev with Windows screen narrator and have it read out the system info like "links" in Swedish and the page content in English, but after fiddling a bit I found out that to get this to work I have to:

1. Got to settings and search for the screen reader settings ("Skärmläsare" in swedish, I think it is called the screen narrator in English but I don't know). If you don't find it via search, try to browse via accessibility settings.
2. Tap on the option to add voices.
![bild](https://github.com/flutter/flutter/assets/4630670/1b9893a9-4a02-4233-beed-944182e8038f)
3. Scroll down and tap on the plus symbol to add voices
![bild](https://github.com/flutter/flutter/assets/4630670/d774fbc9-c589-4734-8ada-6092f994f75e)
4. Select English (USA) and add it in addition to the system locale that is already installed with Windows (in my case Swedish)

Now, when I go to https://flutter.dev in Firefox, it does read out things like link, heading level etc. in Swedish and the actual website content using English (American) voice. 

I have also tried to reproduce with NVDA, but I haven't been able to find a voice configuration that allow it shift between Swedish and Engilsh voices even if it does have an option that seem to be that it should respect the annotated locale of content.

**Comment by loic-sharma** [comment]
Supporting this on Windows requires [IAccessible2](https://wiki.linuxfoundation.org/accessibility/iaccessible2/textattributes) or [UI Automation](https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-automation-element-propids) as Microsoft Active Accessibility does not appear to support language text attributes. In other words, Windows support is blocked on https://github.com/flutter/flutter/issues/94782 and https://github.com/flutter/flutter/issues/127125.

**Comment by whitemagehealing** [comment]
I'm not sure if this should be it's own issue, but I experience the lack of language override also on the Android Emulator Pixel 3a API 34 with Google Play.

I have set the system language to Japanese, and debug deployment still reads the following two items with the Japanese voice versus English. Or am I just using these both wrong? (The code below only has what I believe is the pertinent information.)

```dart
RichText(
      text: TextSpan(
            locale: Locale('en', 'US'),
            semanticsLabel: 'pronounced as '
```

```dart

Text(
      locale: Locale('en', 'US'),
      semanticsLabel: 'pronounced as',
```

**Comment by loic-sharma** [comment]
@whitemagehealing That's weird, I believe that should work on Android. Please create a new issue so that it's routed to the Android team.

The embedder API was added, desktop platforms should now be able expose this information to assistive technologies. Recap of the current state:

Platform | Status | Notes |
-- | -- | --
Linux | ⚠️ | Needs investigation /cc @robert-ancell  
macOS | ✅ | Done as macOS does not support localized string attributes
Web | ⚠️ | Needs investigation /cc @mdebbar 
Windows | ⚠️ | Blocked by UI Automation support (see https://github.com/flutter/flutter/issues/94782, https://github.com/flutter/flutter/issues/127125)

**Comment by Leffe108** [comment]
> I'm not sure if this should be its own issue, but I experience the lack of language override also on the Android Emulator Pixel 3a API 34 with Google Play.
> 
> I have set the system language to Japanese, and debug deployment still reads the following two items with the Japanese voice versus English. The English is said, but with a Japanese accent, so there is no translation to Japanese (which I don't want anyway). Or am I just using these both wrong? (The code below only has what I believe is the pertinent information.)
> 
> ```dart
> RichText(
>       text: TextSpan(
>             locale: Locale('en', 'US'),
>             semanticsLabel: 'pronounced as '
> ```
> 
> ```dart
> Text(
>       locale: Locale('en', 'US'),
>       semanticsLabel: 'pronounced as',
> ```

[locale property of TextSpan](https://api.flutter.dev/flutter/painting/TextSpan/locale.html) is documented to set the assistive tech locale. 

So I believe it should work although, in my reproduction tests I only text property and not semanticsLabel. I don't know if it matters for propagating locale to semanticsLabel, but Semantics widget for example do not support locale property ([I think it should](https://github.com/flutter/flutter/issues/124889), but I agree that it is a later stage to work on)

[locale property of Text](https://api.flutter.dev/flutter/widgets/Text/locale.html) on the other hand is not. Its documented purpose is to pick the right font.

**Comment by robert-ancell** [comment]
I hadn't noticed this, I'm haven't looked into it. The linked PR didn't change anything in this regard.

**Comment by pedromassango** [comment]
Is there any update on this? I see @chunhtai landed a PR recently, but by reading the code it is not clear wether this would make it possible to update the lang attribute of the page.

@chunhtai are you planning to cover that use-case as well?

**Comment by chunhtai** [comment]
This is not done yet, and is the last item remaining
